Source/DafnyPipeline.Test/Boogie.cs

```diff
@@ -0,0 +1,219 @@
+    var h := 0;
+  }
+  
+  var s := {2,3};
+  modify heap {
+    var z := 3; // modify does not push a Dafny scope, so we can't assign to a fresh x here.
+  }
+  
+  assert x == 0 by {
+    var i := x + 2;
+    var j := 0;
+  }
+  var y := 200;
+}";
+          var tempPath = Path.GetTempFileName() + ".dfy";
+          var beforeBoogie = GetBoogie(beforeChange, tempPath);
+          var afterBoogie = GetBoogie(afterChange, tempPath);
+          var diff = GetDiff(beforeBoogie, afterBoogie);
+
+          var expectedDiff = @"-   var eg$y#0: int;
+-   var y#0_0: int;
+-   var x#1_0: int;
+-   var y#2_0: int;
++   var eg$k#0: int;
++   var k#0_0: int;
++   var a#1_0: int;
++   var b#2_0: int;
+-   var x#3_0: int;
+-   var y#3_0: int;
++   var c#3_0: int;
++   var d#3_0: int;
+-   var x#4_0: int;
+-   var y#4_0: int;
++   var e#4_0: int;
++   var f#4_0: int;
+-   var x#5_0: int;
+-   var y#5_0: int;
++   var g#5_0: int;
++   var h#5_0: int;
+-   var x#7_0: int;
+-   var y#7_0: int;
++   var i#7_0: int;
++   var j#7_0: int;
+-     havoc eg$y#0;
++     havoc eg$k#0;
+-     havoc y#0_0;
++     havoc k#0_0;
+-         assume y#0_0 < x#0;
++         assume k#0_0 < x#0;
+-         x#1_0 := LitInt(0);
++         a#1_0 := LitInt(0);
+-         assume !(exists eg$y#1: int :: eg$y#1 < x#0);
++         assume !(exists eg$k#1: int :: eg$k#1 < x#0);
+-         y#2_0 := LitInt(0);
++         b#2_0 := LitInt(0);
+-         x#3_0 := LitInt(2);
++         c#3_0 := LitInt(2);
+-         y#3_0 := LitInt(0);
++         d#3_0 := LitInt(0);
+-         x#4_0 := LitInt(2);
++         e#4_0 := LitInt(2);
+-         y#4_0 := LitInt(0);
++         f#4_0 := LitInt(0);
+-         x#5_0 := LitInt(2);
++         g#5_0 := LitInt(2);
+-         y#5_0 := LitInt(0);
++         h#5_0 := LitInt(0);
+-         x#7_0 := x#0 + 2;
++         i#7_0 := x#0 + 2;
+-         y#7_0 := LitInt(0);
++         j#7_0 := LitInt(0);
+";
+          Assert.Equal(expectedDiff, diff);
+        }
+
+        string GetDiff(string before, string after) {
+          var diff = InlineDiffBuilder.Diff(before, after);
+          var result = new StringBuilder();
+          foreach (var line in diff.Lines)
+          {
+            switch (line.Type)
+            {
+              case ChangeType.Inserted:
+                result.Append("+ ");
+                result.AppendLine(line.Text);
+                break;
+              case ChangeType.Deleted:
+                result.Append("- ");
+                result.AppendLine(line.Text);
+                break;
+            }
+
+          }
+
+          return result.ToString();
+        }
+
+        static Boogie() {
+          var testAssemblyPath = Assembly.GetAssembly(typeof(Boogie)).GetAssemblyLocation();
+          var parts = testAssemblyPath.Split(Path.DirectorySeparatorChar);
+          // This way of finding the repository root is not reliable, we should instead reference the DafnyPipeline assembly and run Dafny in the same process as the unit tests.
```

Comment:
Yup I hit this problem in #680 as well, but thanks for at least coming up with a less horrible way of locating the source. I got close to getting this to work even with a separate CLI process, and will apply the solution here too if I figure it out.